packages/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -0,0 +1,228 @@
+part of firebase_remote_config;
+
+/// The entry point for accessing Remote Config.
+///
+/// You can get an instance by calling [RemoteConfig.instance]. Note
+/// [RemoteConfig.instance] is async.
+class RemoteConfig extends ChangeNotifier {
+  static const MethodChannel _channel =
+      const MethodChannel('plugins.flutter.io/firebase_remote_config');
+
+  static const String defaultValueForString = '';
+  static const int defaultValueForInt = 0;
+  static const double defaultValueForDouble = 0.0;
+  static const bool defaultValueForBool = false;
+
+  static const String _fetchFailedThrottledKey = 'FETCH_FAILED_THROTTLED';
+  static const String _lastFetchTimeKey = 'LAST_FETCH_TIME';
+  static const String _lastFetchStatusKey = 'LAST_FETCH_STATUS';
+  static const String _parametersKey = 'PARAMETERS';
+
+  Map<String, RemoteConfigValue> _parameters;
+
+  DateTime _lastFetchTime;
+  LastFetchStatus _lastFetchStatus;
+  RemoteConfigSettings _remoteConfigSettings;
+
+  RemoteConfig._() {
+    _channel.setMethodCallHandler((MethodCall call) async {
+      switch (call.method) {
+        default:
+          throw new MissingPluginException(
+            '${call.method} method not implemented on the Dart side.',
+          );
+      }
+    });
+  }
+
+  DateTime get lastFetchTime => _lastFetchTime;
+  LastFetchStatus get lastFetchStatus => _lastFetchStatus;
+  RemoteConfigSettings get remoteConfigSettings => _remoteConfigSettings;
+
+  /// Gets the instance of RemoteConfig for the default Firebase app.
+  static Future<RemoteConfig> get instance async {
```

Comment:
I think we need to make the `RemoteConfig` instance a singleton stored in a private static member. Otherwise you could have multiple `RemoteConfig` objects with different default values because they don't know how to synchronize their copies of` _parameters`.

Reply:
Done